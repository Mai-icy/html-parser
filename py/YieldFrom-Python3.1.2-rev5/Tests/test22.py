#
#  Test send and return with value
#

def f(r):
	gi = g(r)
	next(gi)
	try:
		print("f sending spam to g")
		gi.send("spam")
		print("f SHOULD NOT BE HERE")
	except StopIteration as e:
		print("f caught", repr(e))

def g(r):
	print("g starting")
	x = yield
	print("g received", x)
	print("g returning", r)
	return r

f(None)
f(42)
