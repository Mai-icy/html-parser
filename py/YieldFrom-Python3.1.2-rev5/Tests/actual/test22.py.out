g starting
f sending spam to g
g received spam
g returning None
f caught StopIteration()
g starting
f sending spam to g
g received spam
g returning 42
f caught StopIteration(42,)
